### PR TITLE
CORE-2021 - Add map component

### PR DESCRIPTION
**Pull request by NathanAB**
changes: _rev_

### Proposed Changes
_Please use the Jira Key or NOREF followd by the changes_

- CORE-2021
  - Add policy map component

### Checklist

- [ ] I have added or updated unit tests
- [ ] I have added or updated integration tests (if appropriate)
- [ ] I have added or updated documentation / readme (if appropriate)
- [ ] I have verified that my changes have not introduced new lint errors
- [ ] I have updated the change log

### CODE REVIEW

File: lib/components/PolicyMap/PolicyMap.css [hunk 1]
```diff
@@ -0,0 +1,82 @@
+.container {
+  width: 100%;
+  max-width: 935px;
+  height: 539px;
+  cursor: pointer;
+  border: 1px gray solid;
```
Comment (suchak1) [inline lib/components/PolicyMap/PolicyMap.css:6]:
```suggestion
  border: 1px var(--vds-color-slate-darker) solid;
  border-radius: 5px;
```

**Review by suchak1 — COMMENTED** [review]
nit but otherwise v cool

File: lib/components/PolicyMap/PolicyMap.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    // Color the country according to quartile if none is selected
+    if (!focusQuartile) {
+      return { ...dataPoint, color: COLOR_SCALE[colorIndex] };
+    }
+
+    // Is a quartile is selected, only color the country if it falls in the quartile range
+    if (focusQuartile === colorIndex + 1) {
+      return { ...dataPoint, color: COLOR_SCALE[colorIndex] };
+    }
+    return {};
```
Comment (kuguarpwnz) [inline lib/components/PolicyMap/PolicyMap.js:67]: is it intentional? tooltip does not show on inactive countries when we return empty object here. if we do want to always show a tooltip, we should return `defaultPoint`

File: lib/components/PolicyMap/PolicyMap.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  const toggleQuartile = (quartileClicked) => {
+    onFocusQuartile(focusQuartile === quartileClicked ? 0 : quartileClicked);
+  };
+  const [tooltip, setTooltip] = useState();
+
+  let sorted = data.sort((a, b) => {
+    if (a.value < b.value) return -1;
+    if (a.value > b.value) return 1;
+    return 0;
+  });
```
Comment (kuguarpwnz) [inline lib/components/PolicyMap/PolicyMap.js:42]:
```suggestion
  let sorted = data.sort((a, b) => a.value - b.value);
```

File: lib/components/PolicyMap/PolicyMap.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+  // Breakpoints used for legend
+  const breakpoints = [
+    numeral(sorted[Math.floor(0.25 * sorted.length)].value).format('0a'),
+    numeral(sorted[Math.floor(0.25 * sorted.length)].value + 1).format('0a'),
+    numeral(sorted[Math.floor(0.5 * sorted.length)].value).format('0a'),
+    numeral(sorted[Math.floor(0.5 * sorted.length)].value + 1).format('0a'),
+    numeral(sorted[Math.floor(0.75 * sorted.length)].value).format('0a'),
+    numeral(sorted[Math.floor(0.75 * sorted.length)].value + 1).format('0a'),
+    numeral(sorted[sorted.length - 1].value).format('0a'),
+  ];
```
Comment (kuguarpwnz) [inline lib/components/PolicyMap/PolicyMap.js:53]:
```suggestion
const breakpoints = [
    sorted[Math.floor(0.25 * sorted.length)].value,
    sorted[Math.floor(0.25 * sorted.length)].value + 1,
    sorted[Math.floor(0.5 * sorted.length)].value,
    sorted[Math.floor(0.5 * sorted.length)].value + 1,
    sorted[Math.floor(0.75 * sorted.length)].value,
    sorted[Math.floor(0.75 * sorted.length)].value + 1,
    sorted[sorted.length - 1].value,
  ].map((number) => numeral(number).format('0a'));
```

File: lib/components/PolicyMap/PolicyMap.js [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+          scale: 125,
+        }}
+      >
+        <ZoomableGroup zoom={1}>
+          <Geographies geography={geoData}>{mapContent}</Geographies>
+        </ZoomableGroup>
+      </ComposableMap>
+      <legend className={styles.legendContainer}>
+        <ul className={styles.legendList}>
+          <li className={styles.legendListItem}>
```
Comment (kuguarpwnz) [inline lib/components/PolicyMap/PolicyMap.js:140]: we can simplify things a little bit with these changes:
```jsx
const format = (value) => numeral(value).format('0a');
const pair = (result, value, index, array) => {
  if (index % 2 === 0) {
    result.push(array.slice(index, index + 2));
  }

  return result;
};

// Breakpoints used for legend
const breakpoints = [
  1,
  sorted[Math.floor(0.25 * sorted.length)].value,
  sorted[Math.floor(0.25 * sorted.length)].value + 1,
  sorted[Math.floor(0.5 * sorted.length)].value,
  sorted[Math.floor(0.5 * sorted.length)].value + 1,
  sorted[Math.floor(0.75 * sorted.length)].value,
  sorted[Math.floor(0.75 * sorted.length)].value + 1,
  sorted[sorted.length - 1].value,
]
  .map(format)
  .reduce(pair, []);
```

and 

```jsx
{breakpoints.map((breakpointsPair, index) => (
  // eslint-disable-next-line react/no-array-index-key
  <li key={index} className={styles.legendListItem}>
    <button
      className={cn(
        styles.legendListButton,
        focusQuartile && focusQuartile !== index + 1 && styles.legendListDisabled,
      )}
      onClick={() => toggleQuartile(index + 1)}
      type="button"
    >
      <div className={cn(styles.legendBox, styles[`quartile${index + 1}`])} />
      <span>{breakpointsPair.join(' - ')}</span>
    </button>
  </li>
))}
```

**Review by kuguarpwnz — APPROVED** [review]
👍 lgtm

File: lib/components/PolicyMap/PolicyMap.css (hunk 1 above)
Comment (NathanAB) [reply lib/components/PolicyMap/PolicyMap.css:6]: ah good catch

File: lib/components/PolicyMap/PolicyMap.js (hunk 2 above)
Comment (NathanAB) [reply lib/components/PolicyMap/PolicyMap.js:67]: yeah the goal was to completely disable countries outside of active quartile

File: lib/components/PolicyMap/PolicyMap.js (hunk 5 above)
Comment (NathanAB) [reply lib/components/PolicyMap/PolicyMap.js]: good idea, done

File: lib/components/PolicyMap/PolicyMap.css (hunk 1 above)
Comment (suchak1) [reply lib/components/PolicyMap/PolicyMap.css]: @NathanAB i see u changed the border color, just a reminder to add the border-radius line too if u haven't already

Comment (NathanAB) [reply lib/components/PolicyMap/PolicyMap.css]: oh oops, missed that. done

**Comment by virtru-github-public** [comment]
:rocket: PR was released in [`v1.14.0`](https://github.com/virtru/virtuoso-design-system/releases/tag/v1.14.0) :rocket:

**Pull request merged**